added solutions

engineer027 opened the pull request.

engineer027 committed:
added solutions

Sofasmile commented on src/main/java/mate/academy/dao/impl/CarDaoImpl.java:
```diff
@@ -0,0 +1,58 @@
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Car> cars = new ArrayList<>();
```
як варіант можна спробувати зробити це за допомогою стрімів

engineer027 replied:
Бодя говорив на лекціїї, що якщо цикл в циклі то краще використовувати for

Sofasmile commented on src/main/java/mate/academy/model/Car.java:
```diff
@@ -28,8 +28,8 @@ public void setManufacturer(Manufacturer manufacturer) {
 
-    public void setDrivers(List<Driver> drivers) {
-        this.drivers = drivers;
+    public void setDrivers(Driver driver) {
```
давай повернемо до попереднього вигляду) А якщо хочеш, то можна додати окремий метод щоб додавати по одній машині, але в цьому немає особливого сенсу, адже ми це можемо робити на рівні сервісу

Sofasmile commented on src/main/java/mate/academy/service/mpl/CarServiceImpl.java:
```diff
@@ -0,0 +1,55 @@
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        car.getDrivers().add(driver);
```
варто викликати метод update для машини для того щоб зміни порапили в БД

Sofasmile commented on src/main/java/mate/academy/service/mpl/CarServiceImpl.java:
```diff
@@ -0,0 +1,55 @@
+
+    @Override
+    public void removeDriverFromCar(Driver driver, Car car) {
+        car.getDrivers().removeIf(p -> p.equals(driver));
```
the same here

Sofasmile replied:
```suggestion
        car.getDrivers().remove(driver);
```

engineer027 committed:
fix method:
1. setDrivers;
2. addDriverToCar;
3. removeDriverFromCar;
4. create List<Driver> drivers in Application.
